packages/salesforce-adapter/src/change_validators/unknown_users.ts

```diff
@@ -0,0 +1,137 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { collections } from '@salto-io/lowerdash'
+import {
+  ChangeError, isAdditionOrModificationChange, isInstanceChange, ChangeValidator,
+  InstanceElement, getChangeData,
+} from '@salto-io/adapter-api'
+import { apiName } from '../transformers/transformer'
+import { isInstanceOfType, buildSelectQueries, queryClient } from '../filters/utils'
+import SalesforceClient from '../client/client'
+
+const { awu } = collections.asynciterable
+
+// cf. 'Statement Character Limit' in https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select.htm
+const SALESFORCE_MAX_QUERY_LEN = 100000
+
+type GetUserField = (instance: InstanceElement) => string | undefined
+
+type UserFieldGetter = {
+  field: string
+  getter: GetUserField
+}
+
+type TypesWithUserFields = {
+  [type: string]: UserFieldGetter[]
```

Comment:
Nit: we can make this type even more strict. 


```suggestion
 type TypeWithUserField = 'CaseSettings'
  [type: TypeWithUserField]: UserFieldGetter[]
```

Reply:
Typescript question: Why is `[type: TypeWithUserField]: UserFieldGetter[]` preferable to `type TypesWithUserFields = Record<TypeWithUsers, UserFieldGetter[]>`?

Reply:
see suggestion above ^^ I hope this helps.

About 1 => not sure I followed your question, but my 2cents here was that if we have a limited possible set of values, it's better to reflect it in the typesystem vs use "string" (Which is infinite)

for example the following function
string => void (can do a lot of things)

And on the other example, this specific function
(str: 'CaseOwnerSettings') => void (can do much less, and I need to test it for example, with much less inputs)

etc etc